pkg/app/app.go

```diff
@@ -51,8 +55,17 @@ func NewApp(config *config.AppConfig) (*App, error) {
 }
 
 func (app *App) Run() error {
-	err := app.Gui.RunWithSubprocesses()
-
+	// before we do anything, we need to check that we have some window space available
+	var err error
+	width := 0
+	for width == 0 {
+		width, _, err = terminal.GetSize(int(os.Stdin.Fd()))
+		if err != nil {
+			return err
+		}
+		time.Sleep(time.Millisecond * 50)
```

Comment:
I am afraid, this may cause an infinite loop unless we are sure that there can be no way that the width will remain `0` indefinitely. One rudimentary way to mitigate this issue is to keep a loop counter, which, when exceeds a threshold (say `20`) the app exits with a message like `Unable to initialize the window`.

Reply:
Good point. On the other hand, the container crashes if the window is of zero size, so a perhaps better way would be to log to stdout that it is waiting for the window to be created?

By the way, can lazydocker be ran without a container and with a zero sized window? If it is not possible either, the above solution seems like a good fit.

Reply:
In my opinion, such logs should go to `STDERR` not `STDOUT`.

Reply:
I've added that in [the latest commit](https://github.com/jesseduffield/lazydocker/pull/36/commits/29ad43c4a4fdbc09d8085c68c820fbde8aa473a2) where it prints to stderr only when the terminal width is still zero after a first 50 milliseconds sleep.